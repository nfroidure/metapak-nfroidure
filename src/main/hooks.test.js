'use strict';

const hooksTransformer = require('./hooks');

describe('main', () => {
  describe('Hooks transformer', () => {
    it('should add pre-commit hooks', () => {
      expect(hooksTransformer({}, {})).toMatchSnapshot();
    });

    it('should let pre-commit hooks empty for child packages', () => {
      expect(hooksTransformer({}, {})).toMatchSnapshot();
    });
  });
});
