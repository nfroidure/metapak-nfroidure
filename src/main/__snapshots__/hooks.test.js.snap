// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main Hooks transformer should add pre-commit hooks 1`] = `
Object {
  "commit-msg": Array [
    "
if [ \\"$NODE_ENV\\" != \\"cli\\" ] ; then
  if grep -q '^[0-9]\\\\+.[0-9]\\\\+.[0-9]\\\\+$' \\"$1\\" ; then
    exit 0;
  else
    echo \\"‚ö†Ô∏è - Please commit with \\\\\`npm run cz -- (usual commit args)\\\\\`\\"
    echo \\"üíä - To bypass commitizen add \\\\\`NODE_ENV=cli\\\\\` to your command\\"
    echo \\"üí° - You may want to set an alias: \\\\\`alias gicz='npm run cz -- '\\\\\`\\"
    exit 1;
  fi
fi",
  ],
  "pre-commit": Array [
    "
if ! git diff-files --quiet --ignore-submodules ; then
  echo \\"‚ö†Ô∏è - Unstaged files found:\\"
  echo $(git diff-files --shortstat)
fi",
  ],
}
`;

exports[`main Hooks transformer should let pre-commit hooks empty for child packages 1`] = `
Object {
  "commit-msg": Array [
    "
if [ \\"$NODE_ENV\\" != \\"cli\\" ] ; then
  if grep -q '^[0-9]\\\\+.[0-9]\\\\+.[0-9]\\\\+$' \\"$1\\" ; then
    exit 0;
  else
    echo \\"‚ö†Ô∏è - Please commit with \\\\\`npm run cz -- (usual commit args)\\\\\`\\"
    echo \\"üíä - To bypass commitizen add \\\\\`NODE_ENV=cli\\\\\` to your command\\"
    echo \\"üí° - You may want to set an alias: \\\\\`alias gicz='npm run cz -- '\\\\\`\\"
    exit 1;
  fi
fi",
  ],
  "pre-commit": Array [
    "
if ! git diff-files --quiet --ignore-submodules ; then
  echo \\"‚ö†Ô∏è - Unstaged files found:\\"
  echo $(git diff-files --shortstat)
fi",
  ],
}
`;
